app/celery/process_ga4_measurement_tasks.py

```diff
@@ -72,6 +55,18 @@ def post_to_ga4(
         current_app.logger.error('GA4_MEASUREMENT_ID is not set')
         return False
 
+    # Retrieve the notification from the database
```

Comment:
This makes more sense - I remember we originally wanted to avoid doing a query in the celery task. But, I don't see/remember this disadvantage of doing a query here. 

It's a pretty simple query too - so I don't see the harm.